### PR TITLE
Add F option for controlling PWM freq to M106

### PR DESCRIPTION
**Pull request by wilhoj**
Small addition for M106 g-code to change PWM frequency.  

should be compliant with reprap g-code style
http://reprap.org/wiki/G-code#M106:_Fan_On

this should be further developed to make it persistent, but that need more changes on code..

### CODE REVIEW

**Comment by wilhoj** [comment]
My problem on first place was that i did not get my small 40x40 or 30x30 fans to run.. 30mm did run on full power (S255), but other that that those only make high pitch noise..

After playing with PWM setting i did get those spinning at S40-60 and linearly speed up till S255

File: redeem/gcodes/M106_M107.py [hunk 1]
```diff
@@ -41,7 +44,8 @@ def get_long_description(self):
         return "Set the current fan power. Specify S parameter for the " \
                "power (between 0 and 255) and the P parameter for the fan " \
                "number. P=0 and S=255 by default. If no P, use fan from config. "\
-               "If no fan configured, use fan 0. If 'R' is present, ramp to the value"
+               "If no fan configured, use fan 0. If 'R' is present, ramp to the value"\
+               "if 'F' present change PWM frequency hz eg. F1000 is 1khz"
```
Comment (ajmirsky) [inline redeem/gcodes/M106_M107.py]: thank you for being thorough and updating the description too!

appreciate the contribution : )

**Comment by wilhoj** [comment]
new commit to add persistence for pwm frequency.
Using 'Cold-ends', 'pwm_freq' on config.
using "1000" as default in default.cfg as it were hardcoded on Redeem.py

File: configs/default.cfg [hunk 2]
```diff
@@ -275,6 +275,9 @@ cooler_0_target_temp = 60
 # different temperature:
 # therm-e-fan-0-target_temp = 70
 
+# Sets pwm frequency used for controlling fans
```
Comment (ThatWileyGuy) [inline configs/default.cfg]: Could you add a note that this only applies to the B* revision boards?

Comment (wilhoj) [reply configs/default.cfg]: did some mention on g-code "help" section

**Comment by wilhoj** [comment]
Added info about revisions

**Pull request merged**